helix-core/src/main/java/org/apache/helix/common/caches/AbstractDataCache.java

```diff
@@ -43,8 +47,13 @@
 
   protected ControlContextProvider _controlContextProvider;
 
+  protected Map<String, Map<String, Map<String, T>>> _participantStateMap;
+
+  protected Map<PropertyKey, T> _participantStateCache = Maps.newHashMap();
+
   public AbstractDataCache(ControlContextProvider controlContextProvider) {
```

Comment:
Thanks for the change. Looks great.
One suggestion is that this change will reduce the genericity of AbstractDataCache. Can we add a middle class called "AbstractParticipantStateCache" as a child class of the AbstractDataCache?